### DISCUSSION TITLE
switchLocalePath triggers useAsyncData watch

### DISCUSSION THREAD

**Discussion by martinsjek** [nuxt-modules/i18n#2763] [General]
### Environment

StackBlitz

### Reproduction

https://stackblitz.com/edit/github-orzjqc-5p8wzh?file=nuxt.config.ts,pages%2Findex.vue

### Describe the bug

When changing the language with switchLocalePath it re-executes useAsyncData function.
Is this how it should work?
How do I change the path but do not trigger useAsyncData if it's not based on locale?

**Comment by BobbieGoede** [comment]
This is expected behaviour, you're switching locale by navigating to a (technically different) page so page `setup` is called again. If you don't want this to happen you can move data fetching outside of the page, in something like a store or layout for example.

**Reply by martinsjek** [reply]
Thanks for clarification :)

**Comment by martinsjek** [comment]
@BobbieGoede 
Is this still the case?
Because if I use useAsyncData() without a key let's say:
const { data } = await useAsyncData(() =>
  getSomeData()
    .then((response) => response.data)
    .catch(() => {}),
);

Then after switching locale, this won't re-execute.

**Reply by BobbieGoede** [reply]
My previous reply in this discussion is no longer valid since the caching optimizations in around Nuxt 3.17.

If you expect unique values across languages when using `useAsyncData` you will need to pass it a key that contains the current locale. I think the snippet I shared in a reply to you on a different issue should work: https://github.com/nuxt-modules/i18n/issues/3599#issuecomment-2886367245, let me know if you have any questions.
